frontend/src/components/filtertable.js

```diff
@@ -160,3 +169,137 @@ export class FilterTable extends React.Component {
     );
   }
 }
+
+
+export class MetaFilter extends React.Component {
+  // TODO Extend this to contain the filter handling functions, and better integrate filter state with FilterTable
+  // https://github.com/ibutsu/ibutsu-server/issues/230
+  static propTypes = {
+    fieldOptions: PropTypes.array,  // could reference constants directly
+    runId: PropTypes.string,  // make optional?
+    setFilter: PropTypes.func,
+    customFilters: PropTypes.array, // more advanced handling of filter objects? the results-aggregator endpoint takes a string filter
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      fieldSelection: null,
+      isFieldOpen: false,
+      isValueOpen: false,
+      valueOptions: [],
+      valueSelections: [],
+    };
+  }
+
+  onFieldToggle = isExpanded => {
+    this.setState({isFieldOpen: isExpanded})
+  };
+
+  onValueToggle = isExpanded => {
+    this.setState({isValueOpen: isExpanded})
+  };
+
+  onFieldSelect = (event, selection) => {
+    this.setState(
+      // clear value state too, otherwise the old selection remains selected but is no longer visible
+      {fieldSelection: selection, isFieldOpen: false, valueSelections: [], valueOptions: [], isValueOpen: false},
+      this.updateValueOptions
+    )
+
+  };
+
+  onValueSelect = (event, selection) => {
+    // update state and call setFilter
+    const valueSelections = this.state.valueSelections;
+    let updated_values = (valueSelections.includes(selection))
+      ? valueSelections.filter(item => item !== selection)
+      : [...valueSelections, selection]
+
+    this.setState(
+      {valueSelections: updated_values},
+      () => this.props.setFilter(this.state.fieldSelection, this.state.valueSelections.join(';'))
+    )
+  };
+
+  onFieldClear = () => {
+    this.setState(
+      {fieldSelection: null, valueSelections: [], isFieldOpen: false, isValueOpen: false},
+    )
+  };
+
+  onValueClear = () => {
+    this.setState(
+      {valueSelections: [], isValueOpen: false},
+      () => this.props.setFilter(this.state.fieldSelection, this.state.valueSelections)
+    )
+  }
+
+  updateValueOptions = () => {
+    const {fieldSelection} = this.state
+    const {customFilters} = this.props
+
+    console.log('CUSTOMFILTER: '+customFilters)
+    if (fieldSelection !== null) {
+      let api_filter = toAPIFilter(customFilters).join()
+      console.log('APIFILTER: '+customFilters)
+
+      fetch(buildUrl(Settings.serverUrl + '/widget/result-aggregator',
```

Comment:
Now that the auth stuff has been merged, this needs to change.

```suggestion
      HttpClient.get([Settings.serverUrl, 'widget', 'result-aggregator'], {...})
          .then(response => HttpClient.handleResponse(response))
          .then(data => ....);
```